pkg/providers/cloudstack/template.go

```diff
@@ -121,10 +121,14 @@ func buildTemplateMapCP(clusterSpec *cluster.Spec) (map[string]interface{}, erro
 		Append(clusterapi.ControlPlaneNodeLabelsExtraArgs(clusterSpec.Cluster.Spec.ControlPlaneConfiguration))
 	apiServerExtraArgs := clusterapi.OIDCToExtraArgs(clusterSpec.OIDCConfig).
 		Append(clusterapi.AwsIamAuthExtraArgs(clusterSpec.AWSIamConfig)).
-		Append(clusterapi.PodIAMAuthExtraArgs(clusterSpec.Cluster.Spec.PodIAMConfig)).
+		Append(clusterapi.APIServerExtraArgs(clusterSpec.Cluster.Spec.ControlPlaneConfiguration.APIServerExtraArgs)).
 		Append(clusterapi.EtcdEncryptionExtraArgs(clusterSpec.Cluster.Spec.EtcdEncryption)).
 		Append(sharedExtraArgs)
-
+	if _, ok := apiServerExtraArgs["service-account-issuer"]; ok && clusterSpec.Cluster.Spec.PodIAMConfig.ServiceAccountIssuer != "" {
+		apiServerExtraArgs["service-account-issuer"] += "," + clusterSpec.Cluster.Spec.PodIAMConfig.ServiceAccountIssuer
+	} else {
+		apiServerExtraArgs.Append(clusterapi.PodIAMAuthExtraArgs(clusterSpec.Cluster.Spec.PodIAMConfig))
```

Comment:
And looks like we don't have a for loop here, this is confusing compared to the above comment I made.

Reply:
There is a for loop here as well. It's hidden within the `Append()` function